test/e2e/test_pubsub.go

```diff
@@ -24,120 +24,87 @@ import (
+		kngcptesting.WithPubSubTopic(topic))
+	client.CreatePubSubOrFail(eventsPubsub)
 
-	// Delete deferred.
-	defer func() {
-		if err := installer.Do("delete"); err != nil {
-			t.Errorf("failed to create, %s", err)
-		}
-		// Just chill for tick.
-		time.Sleep(10 * time.Second)
-	}()
-
-	if err := client.WaitForResourceReady(client.Namespace, psName, schema.GroupVersionResource{
-		Group:    "events.cloud.google.com",
-		Version:  "v1alpha1",
-		Resource: "pubsubs",
-	}); err != nil {
+	if err := client.Core.WaitForResourceReady(psName, lib.PubsubTypeMeta); err != nil {
 		t.Error(err)
 	}
 }
 
 // PubSubWithTargetTestImpl tests we can receive an event from PubSub. If assertMetrics is set to true, we also assert
 // for StackDriver metrics.
-func PubSubWithTargetTestImpl(t *testing.T, packages map[string]string, assertMetrics bool) {
-	topicName, deleteTopic := makeTopicOrDie(t)
+func PubSubWithTargetTestImpl(t *testing.T, assertMetrics bool) {
+	topicName, deleteTopic := lib.MakeTopicOrDie(t)
 	defer deleteTopic()
 
 	psName := helpers.AppendRandomString(topicName + "-pubsub")
 	targetName := helpers.AppendRandomString(topicName + "-target")
 
-	client := Setup(t, true)
+	client := lib.Setup(t, true)
 	if assertMetrics {
 		client.SetupStackDriverMetrics(t)
 	}
-	defer TearDown(client)
-
-	config := map[string]string{
-		"namespace":  client.Namespace,
-		"topic":      topicName,
-		"pubsub":     psName,
-		"targetName": targetName,
-		"targetUID":  uuid.New().String(),
-	}
-	for k, v := range packages {
-		config[k] = v
-	}
-
-	installer := NewInstaller(client.Dynamic, config,
-		EndToEndConfigYaml([]string{"pubsub_target", "istio"})...)
-
-	// Create the resources for the test.
-	if err := installer.Do("create"); err != nil {
-		t.Errorf("failed to create, %s", err)
-		return
-	}
-
-	// Delete deferred.
-	defer func() {
-		if err := installer.Do("delete"); err != nil {
-			t.Errorf("failed to create, %s", err)
-		}
-		// Just chill for tick.
-		time.Sleep(10 * time.Second)
-	}()
-
-	gvr := schema.GroupVersionResource{
-		Group:    "events.cloud.google.com",
-		Version:  "v1alpha1",
-		Resource: "pubsubs",
-	}
-
-	jobGVR := schema.GroupVersionResource{
-		Group:    "batch",
-		Version:  "v1",
-		Resource: "jobs",
-	}
-
-	if err := client.WaitForResourceReady(client.Namespace, psName, gvr); err != nil {
+	defer lib.TearDown(client)
+
+	// Create a target Job to receive the events.
+	job := resources.TargetJob(targetName, []v1.EnvVar{{
+		Name:  "TARGET",
+		Value: "falldown",
+	}})
+	client.CreateJobOrFail(job, lib.WithServiceForJob(targetName))
+
+	// Create the PubSub source.
+	eventsPubsub := kngcptesting.NewPubSub(psName, client.Namespace,
+		kngcptesting.WithPubSubSink(metav1.GroupVersionKind{
+			Version: "v1",
+			Kind:    "Service"}, targetName),
+		kngcptesting.WithPubSubTopic(topicName))
+	client.CreatePubSubOrFail(eventsPubsub)
+
+	// Create the Istio ServiceEntry.
+	istioServiceEntry := resources.IstioServiceEntry(
```

Comment:
wondering if we are doing this istio thing in eventing as well?
can you remind me why we need it in code, and cannot be done while setting up the cluster in the bash script?

Reply:
No we are not using Istio in eventing test.

I "translated" it directly from the original e2e tests, but after I removed it, the tests were still able to pass except `TestBrokerWithPubSubChannel` (I guess it's because that test is using Serving, which depends on Istio?)

Do you know if the `ServiceEntry` has to be created in the namespace where the test is run? If it does not have to be in the same namespace, we can definitely move it to the bash script.

Reply:
I think it doesn't have to be created in the namespace that the test runs. The only step we do in non-test clusters is to setup the cluster local gateway and things work... It might be doable in the bash script

Reply:
Removed all the relevant code in Go, and created it from the bash script.
Thanks!